### PR TITLE
fix(backend): Add pipeline_id to pipeline_source in Run and RecurringRun

### PR DESCRIPTION
**Pull request by gkcalat**
**Description of your changes:**

- Added `pipeline_id` to `Run` and `RecurringRun` messages
- Tested manually:
   - [x] V1-created runs -> v2 runs with pipeline id
   - [x] V2-created runs -> v2 runs with pipeline id
   - [x] V1-created jobs -> v2 recurring runs with pipeline id
   - [x] V2-created recurring runs -> v2 recurring runs with pipeline id

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: backend/api/v2beta1/recurring_run.proto [hunk 1]
```diff
@@ -85,9 +86,11 @@ message RecurringRun {
   // recurring run. Can be either a pipeline version id, or a pipeline spec.
   oneof pipeline_source {
     // The ID of the pipeline version used for creating runs.
-    string pipeline_version_id = 4;
+    string pipeline_version_id = 4 [deprecated=true];
     // The pipeline spec.
     google.protobuf.Struct pipeline_spec = 5;
+    // Reference to a pipeline version containing pipeline_id and pipeline_version_id.
+    PipelineVersionReference pipeline_version_reference = 18;
```
Comment (chensun) [inline backend/api/v2beta1/recurring_run.proto:93]: 6?

**Review by chensun — CHANGES_REQUESTED** [review]

File: backend/api/v2beta1/recurring_run.proto (hunk 1 above)
Comment (chensun) [reply backend/api/v2beta1/recurring_run.proto:93]: NVM

Comment (gkcalat) [reply backend/api/v2beta1/recurring_run.proto:93]: `6` is used by `runtime_config`. The next number is `18`. Similarly, I used `18` in `run.proto`.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks for the quick fix!

**Pull request merged**